File: test_ex_2.js

function estPalindrome(mot) {
    mot = mot.toLowerCase().replace(" ", "");

    let left = 0;
    let right = mot.length - 1;

    while (left < right) {
        if (mot[left] !== mot[right]) {
            return false;
        }
        left++;
        right--;
    }
    return true;
}

let mot1 = "radar";
let mot2 = "hello";
console.log(estPalindrome(mot1)); // true
console.log(estPalindrome(mot2)); // false

File: test_ex_2.py

mot1 = "Radar"
mot2 = "Hello"

def est_palindrome(mot):
    mot = mot.lower().replace(" ", "")

    left = 0
    right = len(mot) - 1

    while left < right:
        if mot[left] != mot[right]:
            return False
        left += 1
        right -= 1
    
    return True

print(est_palindrome(mot1))
print(est_palindrome(mot2))